Sprint_6

alexpesin opened the pull request.

alexpesin committed:
final version

alexpesin committed:
add jacoco report

alexpesin commented:
Пришлось пересоздавать репозиторий
т.к. в прошлом все очень криво закомитилось, а исправить не получилось

pomegranate282922 commented on src/test/java/com/example/AnimalTest.java:
```diff
@@ -0,0 +1,69 @@
+
+public class AnimalTest {
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
```
Это deprecated метод. Стоит использовать метод assertThrows

alexpesin replied:
public ExpectedException thrown = ExpectedException.none(); - убрал
использую assertThrows

pomegranate282922 commented on src/test/java/com/example/AnimalTest.java:
```diff
@@ -0,0 +1,69 @@
+    //для тренировки
+    @Test
+    public void onceMoreThrowsExceptionOnUnsupportedSexTest() throws Exception {
+        Exception exception = Assert.assertThrows(Exception.class, () -> new Animal().getFood("Всеядное"));
```
Вот это лучший вариант!

pomegranate282922 commented on src/test/java/com/example/CatTest.java:
```diff
@@ -0,0 +1,30 @@
+        List<String> actualList = cat.getFood();
+
+        assertEquals(expectedList, actualList);
+    }
```
Стоит использовать мок.

alexpesin replied:
мок добавил

pomegranate282922 commented on src/test/java/com/example/LionTest.java:
```diff
@@ -0,0 +1,68 @@
+    @Test
+    public void getInvalidSexException() throws Exception {
+        thrown.expectMessage("Используйте допустимые значения пола животного - самей или самка");
+        Feline feline = new Feline();
```
Зачем создавать объект, если уже есть мок.

alexpesin replied:
убрал объект, спасибо))
исправил на assertThrows

pomegranate282922 commented on src/test/java/com/example/LionTest.java:
```diff
@@ -0,0 +1,68 @@
+
+       assertEquals(expectedList, actualList);
+
+    }
```
Нужно везде использовать мок где это уместно.

alexpesin replied:
добавил мок

alexpesin replied:
спасибо за замечания)

pomegranate282922 replied:
И тут везде нужно исправить это замечание

pomegranate282922 reviewed (commented):
Многое поправил, но есть ещё немного замечаний.
Для тестов исключения используй, пожалуйста,  везде assertThrows.
Также прошу использовать моки в cat и lion тестах.

alexpesin committed:
исправления по замечаниям
добавил моки в cat и lion тесты
в тестах исключений везде использую assertThrows

pomegranate282922 commented on src/test/java/com/example/CatTest.java:
```diff
@@ -0,0 +1,34 @@
+
+    @Test
+    public void getSoundTest() {
+        feline = new Feline();
```
Нет, так ты вместо мока опять получил реальный метод. Не нужно делать инициализацию опять.

alexpesin replied:
Динара, попробовал исправить твои замечания
Видимо тема моков труднее чем кажется)

pomegranate282922 commented on src/test/java/com/example/CatTest.java:
```diff
@@ -0,0 +1,34 @@
+
+    @Test
+    public void getFoodTest() throws Exception {
+        feline = new Feline();
```
Везде нужно поправить

pomegranate282922 reviewed (commented):
Все-таки ещё есть одно замечание, которое нужно везде поправить.

alexpesin committed:
исправления по замечаниям 2
добавил  аннотацию @RunWith(MockitoJUnitRunner.class) в CatTest и LionTest
классы
в методах getFoodTest() использовал spy -
feline = Mockito.spy(Feline.class);

pomegranate282922 commented on src/test/java/com/example/CatTest.java:
```diff
@@ -0,0 +1,37 @@
+    @Test
+    public void getFoodTest() throws Exception {
+
+        feline = Mockito.spy(Feline.class);
```
А это зачем? Разве наш нужно шпионить за feline?

alexpesin replied:
Динара, привет! 
спасибо за подсказки)
убрал spy 
добавил стаб Mockito.when(feline.eatMeat()).thenReturn(List.of("Животные", "Птицы", "Рыба"));

pomegranate282922 commented on src/test/java/com/example/LionTest.java:
```diff
@@ -0,0 +1,70 @@
+
+   @Test
+    public void getFoodTest() throws Exception {
+       feline = Mockito.spy(Feline.class);
```
И здесь тоже - шпионы используются только в достаточно редких случаях и тут не тот случай. Здесь нужен мок и стаб.

alexpesin replied:
убрал spy
добавил стаб

pomegranate282922 reviewed (commented):
Вместо шпиона прошу использовать мок и стаб.

alexpesin committed:
исправления по замечаниям 3
1. в классах CatTest и  LionTest убрал spy
2. добавил стаб в методе getFoodTest() класса CatTest
Mockito.when(feline.eatMeat()).thenReturn(List.of("Животные", "Птицы", "Рыба"));
3. добавил стаб в методе getFoodTest() класса LionTest
Mockito.when(feline.getFood("Хищник")).thenReturn(List.of("Животные", "Птицы", "Рыба"));

pomegranate282922 reviewed (approved):
Вот теперь зачёт!
